meta-oe/recipes-devtools/perfetto/perfetto.bb

```diff
@@ -0,0 +1,151 @@
+           git://github.com/protocolbuffers/protobuf.git;branch=3.9.x;protocol=https;destsuffix=git/buildtools/protobuf;name=protobuf \
+           git://chromium.googlesource.com/external/github.com/llvm/llvm-project/libcxx.git;protocol=https;destsuffix=git/buildtools/libcxx;name=libcxx \
+           git://chromium.googlesource.com/external/github.com/llvm/llvm-project/libcxxabi.git;protocol=https;destsuffix=git/buildtools/libcxxabi;name=libcxxabi \
+           git://chromium.googlesource.com/external/github.com/llvm/llvm-project/libunwind.git;protocol=https;destsuffix=git/buildtools/libunwind;name=libunwind \
+           git://chromium.googlesource.com/chromium/llvm-project/compiler-rt/lib/fuzzer.git;protocol=https;destsuffix=git/buildtools/fuzzer;name=fuzzer \
+           git://chromium.googlesource.com/external/github.com/google/benchmark.git;protocol=https;destsuffix=git/buildtools/benchmark;name=benchmark \
+           https://storage.googleapis.com/perfetto/libbacktrace-177940370e4a6b2509e92a0aaa9749184e64af43.zip;subdir=git/buildtools;name=libbacktrace \
+           https://storage.googleapis.com/perfetto/sqlite-amalgamation-3350400.zip;subdir=git/buildtools;name=sqlite \
+           git://github.com/sqlite/sqlite.git;branch=branch-3.32;protocol=https;destsuffix=git/buildtools/sqlite_src;name=sqlite_src \
+           git://github.com/open-source-parsers/jsoncpp;branch=master;protocol=https;destsuffix=git/buildtools/jsoncpp;name=jsoncpp \
+           https://storage.googleapis.com/perfetto/llvm-project-3b4c59c156919902c785ce3cbae0eee2ee53064d.tgz;subdir=git/buildtools;name=llvm-project \
+           git://android.googlesource.com/platform/external/zlib.git;branch=master;protocol=https;destsuffix=git/buildtools/zlib;name=zlib \
+           git://fuchsia.googlesource.com/third_party/linenoise.git;branch=master;protocol=https;destsuffix=git/buildtools/linenoise;name=linenoise \
+           https://storage.googleapis.com/perfetto/gn-linux64-1968-0725d782;subdir=git/buildtools/;name=gn \
+           git://chromium.googlesource.com/chromium/llvm-project/cfe/tools/clang-format.git;protocol=https;destsuffix=git/buildtools/clang-format;name=clang-format \
+           "
+
+# Todo: if toolchain is clang, add clang SRC_URI's, or make Perfetto use Yocto's clang
+
+SRCREV_perfetto = "5bd3f582c075d0d026c5fe0b5e291d34dee0d976"
+SRCREV_googletest = "609281088cfefc76f9d0ce82e1ff6c30cc3591e5"
+SRCREV_protobuf = "6a59a2ad1f61d9696092f79b6d74368b4d7970a3"
+SRCREV_libcxx = "d9040c75cfea5928c804ab7c235fed06a63f743a"
+SRC_URI[libcxx.sha256sum] = "c49ecee61ac94784a08ec9ecd5248512d0899ce88b26a90b9406142d93458c63"
+SRCREV_libcxxabi = "196ba1aaa8ac285d94f4ea8d9836390a45360533"
+SRCREV_libunwind = "d999d54f4bca789543a2eb6c995af2d9b5a1f3ed"
+SRCREV_fuzzer = "debe7d2d1982e540fbd6bd78604bf001753f9e74"
+SRCREV_benchmark = "e991355c02b93fe17713efe04cbc2e278e00fdbd"
+SRCREV_libbacktrace = "177940370e4a6b2509e92a0aaa9749184e64af43"
```

Comment:
You don't need both SRCREV and SRC_URI checksum, right as you're not fetching libbacktrace with git fetcher.

Reply:
It seems BitBake still requires a "SRCREV_libbacktrace" even if it's fetched as a zip package, otherwise the following error appears:
```
ERROR: perfetto-1.0-r0 do_fetch: No checksum specified for '/opt/evb-ast2500/downloads/libbacktrace-177940370e4a6b2509e92a0aaa9749184e64af43.zip', please add at least one to the recipe:
SRC_URI[libbacktrace.sha256sum] = "21ac9a4209f7aeef766c482be53a7fa365063c031c7077e2070b491202983b31"
ERROR: perfetto-1.0-r0 do_fetch: Bitbake Fetcher Error: NoChecksumError('Missing SRC_URI checksum', 'https://storage.googleapis.com/perfetto/libbacktrace-177940370e4a6b2509e92a0aaa9749184e64af43.zip;subdir=git/buildtools;name=libbacktrace')
```

Reply:
Are you sure it was after removing "SRCREV_libbacktrace"? The error looks like SRC_URI[libbacktrace.sha256sum] was removed.

Reply:
I realized the "srcrev" was a left-over typo when I found I could not fetch using the name of their git repository. So yes you are right the error is due to the lack of the SHA256 sum.

I tried all 4 possibilities for the libbacktrace.zip:

1. No sha256sum, no srcrev:
Error "no checksum specified"

2. Only sha256sum, no srcrev:
Fetch OK

3. No sha256sum, only srcrev:
Error "no checksum specified"

4. Both sha256sum and srcrev:
Fetch OK

Also tried all 4 possibilities for the libcxx repository:

1. No sha256sum, no srcrev:


2. Only sha256sum, no srcrev:
Error "please set a valid SRCREV for URI"

3. No sha256sum, only srcrev:
Fetch OK

4. Both sha256sum and srcrev:
Fetch OK

So a SHA256 sum is required for a file URI, while srcrev is required for a repository.

Reply:
Yes, checksums are needed for http(s) fetcher, SRCREVs for git fetcher.